### ISSUE TITLE
mkNeovimPlugin: settings option description _assumes_ `setup` function

### ISSUE THREAD

**Issue by MattSturgeon** [nix-community/nixvim#1695]
90% of neovim plugins use a `require('...').setup` function to setup the plugin with any custom configuration, so we assume the `settings` option should have the description:

```
"Options provided to the `require('${pluginName}').setup` function."
```

Well, mkNeovimPlugin's `mkSettingsOption` wrapper [actually does allow](https://github.com/nix-community/nixvim/blob/7a2a25af02be25987aa43cd681312f4b5ba12317/lib/neovim-plugin.nix#L13) passing a custom `description`, however that [isn't exposed](https://github.com/nix-community/nixvim/blob/7a2a25af02be25987aa43cd681312f4b5ba12317/lib/neovim-plugin.nix#L103-L107) outside of mkNeovimPlugin's internals.

With some minor refactoring, I think mkNeovimPlugin's settingsOption implementation could probably be simplified and inlined a little, while also exposing more options to calling-code.

This issue was originally raised here: https://github.com/nix-community/nixvim/pull/1676#discussion_r1639800598

---

@GaetanLepage I've marked you as the assignee because you mentioned planning to go over the mk*Plugin functions soon, **not** because I'm demanding you work on this :smile:. Of course, you can unassign if you've no interest in this.

**Comment by GaetanLepage** [comment]
> @GaetanLepage I've marked you as the assignee because you mentioned planning to go over the mk*Plugin functions soon, **not** because I'm demanding you work on this 😄. Of course, you can unassign if you've no interest in this.

Don't worry, I have absolutely no problem with being tagged to issues.

That is worth looking at during the refactoring of those helpers.

**Comment by MattSturgeon** [comment]
Fixed in #1758
